spec/honeycomb/propagation/honeycomb_modern_spec.rb

```diff
@@ -75,6 +75,18 @@
       nil,
     ]
   end
+
+  it "handles previously-encoded invalid utf8" do
```

Comment:
Ah, good thinking! Hadn't considered this part of the "changing airplane wings mid-flight". ✈️

Reply:
I'd love to take credit for being super smart and thinking of adding this case ahead of time, but really I was guided to add this because the fix for your original Marshal issue revealed [a bug in the Unmarshal implementation](https://github.com/honeycombio/beeline-ruby/pull/232/commits/3dcc5f60b8a28153ae0755af9321276c9073ecb2).